### PR TITLE
Fix | Improve sendAndRetry

### PR DESCRIPTION
**Pull request by Sammyjo20**
I've also added a PHPStan Github action 😁

### CODE REVIEW

**Review by Sammyjo20 — COMMENTED** [review]
Self review

**Comment by juse-less** [comment]
Regarding the PHPStan reports, it's because I've set the level to max in the default PHPStan config (phpstan.dist.neon).

I intended on fixing everything before release (and the GitHub action).
In the meantime I guess we can set the level to 6, and I can increase it in each PR I add for type hints.
This also makes a lot of sense.
If you set it to 6, I'll bump it to 7 in my callable typehint PR. 🙂

**Review by Sammyjo20 — COMMENTED** [review]
Self review

File: src/Contracts/MockClient.php [hunk 1]
```diff
@@ -104,7 +104,7 @@ public function assertNotSent(string|callable $request): void;
      * Assert JSON data was sent
      *
      * @param string $request
-     * @param array $data
+     * @param array<mixed, mixed> $data
```
Comment (juse-less) [inline src/Contracts/MockClient.php:107]: Keys can only really be `int|string` *1. To type hint array keys, you can use `array-key` for those.

1. unless you use WeakMaps, at which point objects are turned into internal hashes.

Comment (juse-less) [inline src/Contracts/MockClient.php]: Keys can only really be `int|string` *1. To type hint array keys, you can use `array-key` for those.

1. unless you use `\WeakMaps`, at which point objects are turned into internal hashes.

```suggestion
     * @param array<array-key, mixed> $data
```

File: composer.json [hunk 2]
```diff
@@ -64,6 +64,9 @@
         ],
         "test": [
             "./vendor/bin/pest"
+        ],
+        "pstan": [
+            "./vendor/bin/phpstan"
```
Comment (juse-less) [inline composer.json]: I think it's better to be explicit in CI:s, so maybe add `analyse` here.
Replace with `analyze` depending on English/American (PHPStan supports both). 😅 

```suggestion
            "./vendor/bin/phpstan analyse"
```

File: .github/workflows/phpstan.yml [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+        uses: shivammathur/setup-php@v2
+        with:
+          php-version: '8.1'
+          coverage: none
+
+      - name: Install composer dependencies
+        uses: ramsey/composer-install@v2
+
+      - name: Run PHPStan
+        run: ./vendor/bin/phpstan --error-format=github
```
Comment (juse-less) [inline .github/workflows/phpstan.yml]: I think it's better to be explicit in CI:s, so maybe add `analyse` here.
Replace with `analyze` depending on English/American (PHPStan supports both). 😅 

```suggestion
        run: ./vendor/bin/phpstan analyse --error-format=github
```

**Review by Sammyjo20 — COMMENTED** [review]
Self review

**Pull request merged**